### ISSUE TITLE
Connect Multiple Db

### ISSUE THREAD

**Issue by narayannk** [StackExchange/StackExchange.Redis#987]
Hi,

I have a requirement. we are creating a Nuget Package for Redis (since we are using micro service and some of the service need s to use redis. So thought of creating a nuget) which internally use the across the project.We are planned to use stackexchange.Redis. In a single redis server we have around 16 databases. In this we are planning to use each one db for each domain.The problem statement is we have to insists the user to use corresponding DB for his Domain. Is it possible to achieve this.Also we need a single place to know all the different database used for each domain.

Please help me on this.

NK

**Comment by TechMinder** [comment]
If you have created a wrapper then you can ask for database index which can be set when creating a connection. You would need to use namespace to identify which domain is using which database index. If I am not wrong you can't put all the database index in the connection string. You can switch or retrieve the database with GetDatabase call.

https://stackoverflow.com/questions/43177773/do-rediss-getdatabase-method-caches-all-database-tables

**Comment by NickCraver** [comment]
The advice above is good - this is what we do at Stack Exchange. Remember: GetDatabase() isn't a server call - it's a client-side operation for the next command to be issued.

If you're curious how we're doing it, I wrote up a bit in a recent post here: https://nickcraver.com/blog/2019/08/06/stack-overflow-how-we-do-app-caching/

Closing out old issues to cleanup, but if people have more questions I'm happy to reopen!
